### ISSUE TITLE
About the mobile end and low-end models

### ISSUE THREAD

**Issue by xiaxc1** [erichlof/THREE.js-PathTracing-Renderer#40]
Hello brother, I have seen your case. I am also studying this ray tracing (path tracing) function recently. The problem I'm facing now is that the mobile device (apple 6sp) or ASUS laptop (the graphics card is AMD radeon hd45) will show a black screen after it is turned on, as shown in the figure below. Would you like to know if this ray tracing can be used for mobile terminals now? Or when it is used for mobile terminal, do you want the shadow of the object to be softer? Is there any other way to realize the ray in nature besides ray tracing!

**Comment by xiaxc1** [comment]
Sorry, my English is not very good. Thank U

**Comment by erichlof** [comment]
@xiaxc1 
Hello, unfortunately none of the demos work on Apple iOS devices, even the latest iPhones and tablets as of 2019. I was at the Apple store the other day and I navigated to my own project page here on Github and tried to run the demos. The only Apple devices that worked were the Macbook Air, Macbook Pro, and desktop, but not the iPad or iPhone. I believe it is because Webgl 2.0 (which my project relies on) is not supported on Apple's smaller devices,  no matter how modern they are. I'm not sure what the roadmap looks like for Webgl 2.0 support for these devices so I cannot speak on that at this time. But I do hope they add it soon - it would be nice to run the demos on Apple devices on the go!

As for the Asus and lower end laptop devices,  I really cannot speak for them either as I don't have access to try all possible combinations out (like YouTube tech channels do). One of the reasons I made my project for the Web is that wherever Chrome, Firefox,  or Edge (Dev channel for now) is supported AND Webgl 2.0 is also supported,  the demos should run without issues. The only problem I have now is that even on the latest mobile as of 2019 and Webgl 2.0 and a modern browser,  the heavy BVH model demos will still not compile on mobile, in other words a black screen or loss of rendering context.  It is an active area of research for me as I do not even know the cause of the failure. But all the simpler demos without BVH should run perfectly on mobile.

Hope I have answered some of your questions. Let me know if I can help further.  Best of luck to you!
-Erich
